### PR TITLE
Adding pid in subject email

### PR DESCRIPTION
**Pull request by MoffatMore**
## Description of the issue/feature this PR addresses

Linked issue: https://github.com/bhp-lims/bhp.lims/issues/195

## Current behavior before PR
Subject of email doesnt have PID to identify an ar

## Desired behavior after PR is merged
Subject of email contains PID to identify an ar

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bhp/lims/impress/emailview.py [hunk 1]
```diff
@@ -4,13 +4,35 @@
...
+        # string interpolation later
+        # N.B. We need to translate the raw string before interpolation
+        subject = self.context.translate(_("Analysis Results for {}"))
+        self.reports = self.get_reports()
+        for num, report in enumerate(self.reports):
+            # get the linked AR of this ARReport
+            ar = report.getAnalysisRequest()
+            pid = ar.ParticipantID
+            print ar.ParticipantID
+        self.email_subject = subject.format(pid)
```
Comment (xispa) [inline bhp/lims/impress/emailview.py]: Note that if more than one report is published at once, only the PID from the last report will be displayed in email's subject. I suggest to simplify the code like follows:
```python
def __init__(self, context, request):
    super(EmailView, self).__init__(context, request)
    participant_ids = self.get_participant_ids()
    self.email_subject = "Analysis Results for {}".format(", ".join(participant_ids))

def get_participant_ids(self):
    """Participant IDs of the reports to be included in the email
    """    
    ars = map(lambda rep: rep.getAnalysisRequest(), self.get_reports())
    pids = map(lambda ar: ar.ParticipantID, ars)
    return list(set(pids))
```

Comment (MoffatMore) [reply bhp/lims/impress/emailview.py]: ok i did that, thanks

Comment (MoffatMore) [reply bhp/lims/impress/emailview.py]: i have made changes as requested, please review them

File: bhp/lims/impress/emailview.py [hunk 2]
```diff
@@ -4,13 +4,36 @@
...
+        self.context = context
+        self.request = request
+        self.url = self.context.absolute_url()
+
+        self.client_name = safe_unicode(self.context.Title())
+        self.email_body = self.context.translate(_(self.email_template(self)))
+        # string interpolation later
+        # N.B. We need to translate the raw string before interpolation
+        subject = self.context.translate(_("Analysis Results for {}"))
+        participant_ids = self.get_participant_ids()
```
Comment (xispa) [inline bhp/lims/impress/emailview.py]: All lines above this one can be removed, except the call to parent's constructor `super(EmailView, self).__init__(context, request)`

File: bhp/lims/impress/emailview.py (hunk 1 above)
Comment (MoffatMore) [reply bhp/lims/impress/emailview.py]: I have made changes as you requested, thanks i will be waiting for feedback

**Pull request merged**